### PR TITLE
fix(path): remove fromFileUrl

### PR DESCRIPTION
**Pull request by RyanClementsHax**
- replace all usages of fromFileURL within node/polyfills/path to pathFromURL within ext/web/00_infra.js
- remove fromFileURL within node/polyfills/path
- addresses #18177

### CODE REVIEW

File: cli/tests/unit_node/testdata/child_process_stdio.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import process from "node:process";
 import * as path from "node:path";
 
 const script = path.join(
-  path.dirname(path.fromFileUrl(import.meta.url)),
+  path.dirname(new URL(import.meta.url).pathname),
```
Comment (dsherret) [inline cli/tests/unit_node/testdata/child_process_stdio.js:6]: This won't work on Windows. These can be changed to use `fileURLToPath` from `node:url`

Comment (RyanClementsHax) [reply cli/tests/unit_node/testdata/child_process_stdio.js]: thanks! I made the update

File: ext/node/polyfills/process.ts [hunk 2]
```diff
@@ -705,7 +706,7 @@ internals.__bootstrapNodeProcess = function (
   Object.defineProperty(argv, "1", {
     get: () => {
       if (Deno.mainModule.startsWith("file:")) {
-        return fromFileUrl(Deno.mainModule);
+        return pathFromURL(Deno.mainModule);
```
Comment (kt3k) [inline ext/node/polyfills/process.ts:709]: `Deno.mainModule` is a string url, and `pathFromURL` only works for `URL` input (unlike `fromFileUrl`). I think it needs to be converted to URL first

```suggestion
        return pathFromURL(new URL(Deno.mainModule));
```

Comment (RyanClementsHax) [reply ext/node/polyfills/process.ts:709]: Oh! thanks. I'll update. I really appreciate the help :)

Comment (RyanClementsHax) [reply ext/node/polyfills/process.ts]: Thanks for finding this bug. I was so confused why the build was failing in CI when this was working on my local machine lol (using the provided devcontainer).

This brings up the question on why the failing test would pass on my local machine but fail in CI despite the failed step also using a posix OS. 🤔

**Review by kt3k — APPROVED** [review]
LGTM. Nice work!

**Comment by RyanClementsHax** [comment]
Thanks :)

**Pull request merged**